nativepython/tests/string_compilation_test.py

```diff
@@ -457,3 +457,67 @@ def test_find_perf(f, s, subs):
                 .format(c_elapsed, elapsed)
         )
         """
+
+    def check_string_join(self, f):
+        """This function contains tests for implementation of the string join
+        for both: ListOf(str) and TupleOf(str).
+        The tests for the types are as separate functions, so it will be easier
+        to find the failing line later.
+        """
```

Comment:
need some tests with strings that are not just 1 byte per codepoint. Try joining with two-byte and four-byte codepoints in both separator and individual string values.

Reply:
done